sdk/python/feast/infra/contrib/spark_kafka_processor.py

```diff
@@ -116,9 +117,27 @@ def __init__(
         # data_source type has been checked to be an instance of KafkaSource.
         self.data_source: KafkaSource = self.data_source  # type: ignore
 
+    def _create_infra_if_necessary(self):
```

Comment:
Would it be better to name it something like setup_infra_if_lazy_creation?

Reply:
Leaving it for now. We are looking for better ways to manage infra. This is not a sustainable solution.